packages/vite/src/esbuild-resolver.ts

```diff
@@ -0,0 +1,132 @@
+import type { Plugin as EsBuildPlugin, ImportKind, OnResolveResult, PluginBuild } from 'esbuild';
+import { type PluginItem, transform } from '@babel/core';
+import {
+  type Resolution,
+  type ResolverFunction,
+  ResolverLoader,
+  virtualContent,
+  locateEmbroiderWorkingDir,
+} from '@embroider/core';
+import { readFileSync, readJSONSync } from 'fs-extra';
+import { EsBuildModuleRequest } from './esbuild-request';
+import assertNever from 'assert-never';
+import { dirname, resolve } from 'path';
+import { hbsToJS } from '@embroider/core';
+import { Preprocessor } from 'content-tag';
+
+export function esBuildResolver(root = process.cwd()): EsBuildPlugin {
+  let resolverLoader = new ResolverLoader(process.cwd());
+  let macrosConfig: PluginItem | undefined;
+  let preprocessor = new Preprocessor();
+
+  return {
+    name: 'embroider-esbuild-resolver',
+    setup(build) {
+      build.onResolve({ filter: /./ }, async ({ path, importer, pluginData, kind }) => {
+        let request = EsBuildModuleRequest.from(path, importer, pluginData);
+        if (!request) {
+          return null;
+        }
+        let resolution = await resolverLoader.resolver.resolve(request, defaultResolve(build, kind));
+        switch (resolution.type) {
+          case 'found':
+            return resolution.result;
+          case 'not_found':
+            return resolution.err;
+          default:
+            throw assertNever(resolution);
+        }
+      });
+
+      build.onLoad({ namespace: 'embroider', filter: /./ }, ({ path }) => {
+        let src = virtualContent(path, resolverLoader.resolver);
+        if (!macrosConfig) {
+          macrosConfig = readJSONSync(
+            resolve(locateEmbroiderWorkingDir(root), 'rewritten-app', 'macros-config.json')
+          ) as PluginItem;
+        }
+        return { contents: runMacros(src, path, macrosConfig) };
+      });
+
+      build.onLoad({ filter: /\.gjs$/ }, async ({ path: filename }) => {
```

Comment:
should this also handle gts?

or is that automatic somehow?

Reply:
For now I would like to not add typescript to this setup because we might have other considerations that we would need to cover. I would prefer if we added specific support for it in its own PR if that's ok 👍